### PR TITLE
Fix Storage Long Download Pause n' Cancel Integration Test

### PR DESCRIPTION
**Pull request by DellaBitta**
Fixes for the Storage LongDownloadSuspendAndCancel test which was completing too quickly on GHA for the pause and cancel to take place.  Reworked the test to detect when transfers have started instead of waiting xyz milliseconds.

### CODE REVIEW

File: storage/integration_test/src/integration_test.cc [hunk 1]
```diff
@@ -773,4 +788,4 @@ TEST_F(FirebaseStorageTest, TestInvalidatingReferencesWhenDeletingApp) {
   EXPECT_FALSE(metadata.is_valid());
 }
 
-}  // namespace firebase_testapp_automated
+}  // namespace firebase_testapp_automated
```
Comment (vimanyu) [inline storage/integration_test/src/integration_test.cc:791]: nit: no newline at end of file

File: storage/integration_test/src/integration_test.cc [hunk 2]
```diff
@@ -588,7 +588,10 @@ TEST_F(FirebaseStorageTest, TestLargeFilePauseResumeAndDownloadCancel) {
...
     ASSERT_TRUE(controller.is_valid());
 
-    ProcessEvents(500);
+    while(controller.bytes_transferred() == 0) 
+    {
+      ProcessEvents(1);
+    }
     // After waiting a moment for the operation to start, pause the operation
     // and verify it was successfully paused. Note that pause might not take
     // effect immediately, so we give it a few moments to pause before
```
Comment (anonymous-akorn) [inline storage/integration_test/src/integration_test.cc:597]: Do the comments need to be updated to reflect the new strategy?

**Review by vimanyu — COMMENTED** [review]
Looks good except a super minor nit.

File: storage/integration_test/src/integration_test.cc (hunk 2 above)
Comment (DellaBitta) [reply storage/integration_test/src/integration_test.cc:597]: Nope, this is the same behavior as before.

Comment (anonymous-akorn) [reply storage/integration_test/src/integration_test.cc:597]: hmm I guess we're just being smarter about how we're "waiting a moment". Fair.

Comment (DellaBitta) [reply storage/integration_test/src/integration_test.cc:597]: Though I don't think that comment was accurate even in the previous implementation.  updating it.

File: storage/integration_test/src/integration_test.cc (hunk 1 above)
Comment (DellaBitta) [reply storage/integration_test/src/integration_test.cc]: Done!

**Review by anonymous-akorn — APPROVED** [review]

**Review by vimanyu — APPROVED** [review]
LGTM

**Pull request merged**